tests/unit/test_proxy.py

```diff
@@ -16,15 +66,39 @@ def val(self, arg):
             """A test method to check function call on the Temp class."""
             return arg + 123  # A simple operation to check
 
-    def test_attr_simple(self):
-        """Proxy object reads the right input attribute."""
+    def test_attr_simple_action(self):
+        """ActionProxy object reads the right input attribute."""
         node = Node(None, None)
-        proxy = Proxy(node)
+        proxy = ActionProxy(node)
         func = proxy.attr
 
         self.assertEqual(proxy._cur_attr, "attr")
         self.assertTrue(callable(func))
 
+    def test_attr_simple_transformation(self):
+        """
+        TransformationProxy object reads the right input attributes,
+        returning the methods and the attributes of the proxied node.
+        """
+        node = Node(None, None)
+        proxy = TransformationProxy(node)
+
+        for attr in current_backend.supported_operations:
+            getattr(proxy, attr)
+            self.assertEqual(proxy.proxied_node._cur_attr, attr)
+            self.assertTrue(callable(proxy.proxied_node._call_handler))
+
+        proxy.attribute
+        self.assertEqual(proxy.proxied_node._cur_attr, "attribute")
+
+    def test_proxied_node_has_user_references(self):
+        """check that the user reference holds until the proxy lives."""
```

Comment:
Also, try to be consistent with the docstrings, I would start with uppercase all docstrings.